src/main/kotlin/org/rust/lang/core/psi/ext/RsQualifiedNamedElement.kt

```diff
@@ -358,7 +299,16 @@ data class RsQualifiedName private constructor(
 
         private fun RsQualifiedNamedElement.toParentItem(): Item? {
             val name = ((this as? RsMod)?.modName ?: name) ?: return null
-            val itemType = when (this) {
+            if (this is RsMod && isCrateRoot) {
+                val crateName = containingCrate.asNotFake?.normName ?: return null
+                return Item(crateName, CRATE, this)
+            }
```

Comment:
Do we really need to extract this part from `parentItemType`?

It's quite unexpected and can easily lead to incorrect behaviour if user of this code doesn't know about this implementation detail